src/Schema/Schema.php

```diff
@@ -305,6 +480,78 @@ public function findAttribute($name, $namespace = null)
      */
     public function findAttributeGroup($name, $namespace = null)
     {
-        return $this->findSomething('getAttributeGroup', $name, $namespace);
+        /**
+         * @var AttributeGroup
+         */
+        $out = $this->findSomething('getAttributeGroup', $name, $namespace);
+
+        return $out;
+    }
+
+    /**
+     * @var Schema[]
+     */
+    protected static $loadedFiles = array();
+
+    /**
+     * @param string ...$keys
+     *
+     * @return bool
+     */
+    public static function hasLoadedFile(...$keys)
```

Comment:
Can this `(has|get|set)LoadedFile` be moved to `SchemaReader`?

Reply:
ah, I missed these; I was mostly checking for `public static function` where `SchemaReader` was a parameter. will get them tonight.

Reply:
sorted in 8491305